### ISSUE TITLE
Proposal: introduce a DataObject::getCMSTitle() method

### ISSUE THREAD

**Issue by kinglozzer** [silverstripe/silverstripe-framework#9214]
Currently the CMS relies on [`DataObject::getTitle()`](https://github.com/silverstripe/silverstripe-framework/blob/4.4.3/src/ORM/DataObject.php#L879-L907) which looks at the following DB fields to generate a title to be used in breadcrumbs/toast messages, in order:

1. `Title`
2. `Name`
3. `#<id>`

Sometimes that method can’t generate an appropriate title - for example, if `Title` is a `DBText` and the value is split over multiple lines, the breadcrumbs display across multiple lines too. Overriding `getTitle()` also has implications for how the value of a DB field named `Title` is populated in edit forms etc.

I’m proposing we introduce the following method to `DataObject` and use it for all breadcrumb/toast/confirmation messages:

```php
/**
 * Generate a human-readable title for this DataObject to be displayed in the CMS
 *
 * @return string
 */
public function getCMSTitle(): string
{
    return $this->getTitle();
}
```

There should be no behaviour change for existing sites, but it adds the ability to customise the title that’s shown in the CMS for an object. Note that `GridField` can display non-DataObjects, so we’d need to use something like:

```php
if ($this->record->hasMethod('getCMSTitle')) {
    $title = $this->record->getCMSTitle();
} else {
    $title = ($this->record->Title) ? $this->record->Title : "#{$this->record->ID}";
}
```

**Comment by sminnee** [comment]
Yeah I like this. Somewhat related I would also suggest that we promote the use of `CMSEditLink()` to return a default link for editing a given record in the CMS.
